includes/Classifai/Features/Smart404EPIntegration.php

```diff
@@ -0,0 +1,681 @@
+<?php
+/**
+ * Integration with ElasticPress for the Smart 404 Feature.
+ */
+
+namespace Classifai\Features;
+
+use Classifai\Features\Smart404;
+use Classifai\Providers\OpenAI\Tokenizer;
+use ElasticPress\Indexables;
+use ElasticPress\Indexable;
+use ElasticPress\Elasticsearch;
+use WP_Error;
+use WP_CLI;
+
+use function ElasticPress\Utils\is_indexing_wpcli;
+
+/**
+ * ElasticPress Integration class.
+ */
+class Smart404EPIntegration {
+
+	/**
+	 * Embeddings handler.
+	 *
+	 * @var Embeddings $embeddings_handler
+	 */
+	protected $embeddings_handler;
+
+	/**
+	 * Elasticsearch version.
+	 *
+	 * @var string $es_version
+	 */
+	protected $es_version;
+
+	/**
+	 * Tokenizer handler.
+	 *
+	 * @var Tokenizer $tokenizer
+	 */
+	protected $tokenizer;
+
+	/**
+	 * Embeddings meta key.
+	 *
+	 * @var string $embeddings_meta_key
+	 */
+	protected $embeddings_meta_key = '';
+
+	/**
+	 * Post content hash meta key.
+	 *
+	 * @var string $content_hash_meta_key
+	 */
+	protected $content_hash_meta_key = 'classifai_post_content_hash';
+
+	/**
+	 * Setup needed variables.
+	 *
+	 * @param Classifai\Providers\Provider $provider Provider to use for embeddings.
+	 */
+	public function __construct( $provider = null ) {
+		$this->embeddings_handler = $provider;
+		$this->es_version         = ! $provider ? '7.0' : Elasticsearch::factory()->get_elasticsearch_version();
+		$this->tokenizer          = ! $this->embeddings_handler ? new Tokenizer( 8191 ) : new Tokenizer( (int) $this->embeddings_handler->get_max_tokens() );
+
+		if ( $provider ) {
+			if ( 'openai_embeddings' === $provider::ID ) {
+				$this->embeddings_meta_key = 'classifai_openai_embeddings';
+			} elseif ( 'azure_openai_embeddings' === $provider::ID ) {
+				$this->embeddings_meta_key = 'classifai_azure_openai_embeddings';
+			}
+		}
+	}
+
+	/**
+	 * Inintialize the class and register the needed hooks.
+	 */
+	public function init() {
+		// Vector support was added in Elasticsearch 7.0.
+		if ( version_compare( $this->es_version, '7.0', '<=' ) ) {
```

Comment:
```suggestion
		if ( ! $this->es_version || version_compare( $this->es_version, '7.0', '<=' ) ) {
```

As this gives PHP deprecated notice if Elasticsearch is not configured.
`PHP Deprecated:  version_compare(): Passing null to parameter #1 ($version1) of type string is deprecated in /Users/****/wp-content/plugins/classifai/includes/Classifai/Features/Smart404EPIntegration.php on line 82`